app/(header-default)/documents/[slug]/_components/carte-professionnelle-TP-section.tsx

```diff
@@ -5,15 +5,17 @@ import { Icon } from '#components-ui/icon/wrapper';
 import { AsyncDataSectionClient } from '#components/section/data-section/client';
 import { EAdministration } from '#models/administrations/EAdministration';
 import { IUniteLegale } from '#models/core/types';
-import useFetchCarteProfessionnelleTP from 'hooks/fetch/carte-professionnelle-TP';
+import { useAPIRouteData } from 'hooks/fetch/use-API-route-data';
 
 export default function CarteProfessionnelleTPSection({
   uniteLegale,
 }: {
   uniteLegale: IUniteLegale;
 }) {
-  const carteProfessionnelleTravauxPublics =
-    useFetchCarteProfessionnelleTP(uniteLegale);
+  const carteProfessionnelleTravauxPublics = useAPIRouteData(
+    'espace-agent/carte-professionnelle-TP',
```

Comment:
The best would be either to call and enum or another typed object with auto completion rather than a string

Reply:
Autocompletion is enabled with `string`, without the need for importing a data object.